### PR TITLE
Add support to build & publish ARM & ppc64le wheels

### PR DESCRIPTION
**Pull request by mgiessing**
Signed-off-by: Marvin Giessing <marvin.giessing@gmail.com>

Thanks for this nice library @amitdev !
As this currently does not produce IBM Power wheels I've enabled cibuildwheel to support multi arch wheels (aarch64 & ppc64le) - this is not a change contentwise.

### CODE REVIEW

**Comment by amitdev** [comment]
Thanks. Looks good to me, but tbh I'm not familiar with cibuildwheel. Will wait to see if @AndreMiras has any comments before merging this.

File: .github/workflows/build-and-deploy.yml [hunk 1]
```diff
@@ -12,12 +12,16 @@ jobs:
...
+          platforms: all
       # Used to host cibuildwheel
-      - uses: actions/setup-python@v3
-      - name: Install cibuildwheel
-        run: python -m pip install cibuildwheel==2.7.0
       - name: Build wheels
-        run: python -m cibuildwheel --output-dir wheelhouse
+        uses: pypa/cibuildwheel@v2.7.0
+        env:
+          CIBW_ARCHS_LINUX: auto aarch64 ppc64le
```
Comment (AndreMiras) [inline .github/workflows/build-and-deploy.yml:24]: We could risk it a little comment for non cibuildwheel experts.
Something like they have in the doc:
```
# configure cibuildwheel to build native archs ('auto'), and some emulated ones
```

**Review by AndreMiras — APPROVED** [review]
Thanks for the change!
Left a few comments and questions, but looks overall good to me.
Do you mind removing the `.DS_Store` file from that PR and maybe adding it to the `.gitignore`.
Another thing I noticed is the [build](https://github.com/amitdev/lru-dict/actions/runs/3882528202) is giving the following warning:
> The macOS-10.15 environment is deprecated, consider switching to macos-11(macos-latest), macos-12 instead

We cam argue this is out of the scope, but it could be nice to address, I leave that up to you

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
       - run: pytest
 
   install:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (mgiessing) [reply .github/workflows/tests.yml:21]: Leaving this on ubuntu-latest caused fails during test (I think it was python3.6 but I cannot recap 100%).
As the main intention was to enable multiarch builds I didn't wanted to update too much aside of that (except for the changes to make sure the tests succeed).
If that is fine for you I can update it to 22.04

File: .github/workflows/build-and-deploy.yml (hunk 1 above)
Comment (mgiessing) [reply .github/workflows/build-and-deploy.yml:24]: Sure - I'll put that into the .yml :-)

File: .github/workflows/tests.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndreMiras) [reply .github/workflows/tests.yml:21]: OK thanks for the context.
22.04 may behave the same as ubuntu-latest then.
But give it a try at least I'll be able to see the errors and maybe I can help with that.
Regarding Python 3.6 I think it got end of life support last year already, so I would be tempted to drop it if it causes troubles, but that's more up to @amitdev

**Comment by mgiessing** [comment]
just commited all the discussed changes, but it was indeed python3.6 who is the culprit here, since it doesn't seem to be available for ubuntu-22.04

```
0s
Run actions/setup-python@v4
Installed versions
  Version 3.6 was not found in the local cache
  Error: The version '3.6' with architecture 'x64' was not found for Ubuntu 22.04.
  The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
```

We could either remove it as you suggested or I revert that change to back to 20.04

**Comment by AndreMiras** [comment]
Thanks for the details!
Yeah I would rather drop 3.6 than downgrading OS version, but let's see what Amit has in mind.
In the meantime I've created a PR to drop both 2.7 and 3.6 and add 3.11.
https://github.com/amitdev/lru-dict/pull/42

**Comment by mgiessing** [comment]
@amitdev what is your view on that? :-)

**Pull request merged**